package/src/components/MessageOverlay/hooks/useMessageActionAnimation.tsx

```diff
@@ -0,0 +1,40 @@
+import type { ViewStyle } from 'react-native';
+import type { TapGestureHandlerStateChangeEvent } from 'react-native-gesture-handler';
+import {
+  runOnJS,
+  useAnimatedGestureHandler,
+  useAnimatedStyle,
+  useSharedValue,
+} from 'react-native-reanimated';
+import type { MessageActionListItemProps } from '../MessageActionListItem';
+
+export const useMessageActionAnimation = ({
+  action,
+}: Pick<MessageActionListItemProps, 'action'>) => {
+  const opacity = useSharedValue(1);
+
+  const onTap = useAnimatedGestureHandler<TapGestureHandlerStateChangeEvent>(
+    {
+      onEnd: () => {
+        runOnJS(action)();
+      },
+      onFinish: () => {
+        opacity.value = 1;
+      },
+      onStart: () => {
+        opacity.value = 0.2;
```

Comment:
Can we move these values into a couple of constants? or should it be possible to override it somehow? The numbers seem kind of arbitrarily selected the way it looks 🤔

Reply:
Agreed. `1` is fine I would say as it is. But we should introduce `activeOpacity` param to this hook - with default value as 0.2